File: Data Structures & Algorithms/trapping-rain-water/submission-0.java
class Solution {

    public int trap(int[] height) {
        int n = height.length;
        int[] left = new int[n];
        int[] right = new int[n];
        // finding left heights
        int max_height = height[0];
        for (int i = 0; i < n; i++) {
            max_height = Math.max(max_height, height[i]);
            left[i] = max_height;
        }

        // finding right heights
        max_height = height[n - 1];
        for (int i = n - 1; i >= 0; i--) {
            max_height = Math.max(max_height, height[i]);
            right[i] = max_height;
        }

        // finding total logged water blocks
        int water_logged_count = 0;
        for (int i = 1; i < n - 1; i++) {
            water_logged_count += Math.min(left[i], right[i]) - height[i];
        }
        return water_logged_count;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-0.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        left,right,move=0,len(nums)-1,0
        while move<=right:
            if nums[move]==0:
                nums[left],nums[move]=nums[move],nums[left]
                left,move=left+1,move+1
            elif nums[move]==2:
                nums[move],nums[right]=nums[right],nums[move]
                right-=1
            else:
                move+=1
        

